### PR TITLE
Chromatic telemetry was reporting null - Add serverRequire to interpret files

### PR DESCRIPTION
**Pull request by JonathanKolnik**
There are cases where 
```  
const r = typeof __non_webpack_require__ !== 'undefined' ? __non_webpack_require__ : require;
```
still is undefined

so now we fallback to the way storybook interprets these modules

❓ should we just always do the storybook pattern?

### CODE REVIEW

File: bin-src/lib/interpret-require.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+// const registered = false;
+
+export function interopRequireDefault(filePath: string) {
+  // // eslint-disable-next-line no-underscore-dangle, global-require
+  // const hasEsbuildBeenRegistered = !!require('module')._extensions['.ts'];
+
+  // if (registered === false && !hasEsbuildBeenRegistered) {
+  //   // eslint-disable-next-line global-require
+  //   const { register } = require('esbuild-register/dist/node');
```
[the remark this answers is not in the document]
Comment (thafryer) [reply bin-src/lib/interpret-require.ts:12]: Probably since esbuild is not a CLI dependency

[the remark this answers is not in the document]
Comment (JonathanKolnik) [reply bin-src/lib/interpret-require.ts:12]:
```
[bundle:bin   ] ERROR in ./node_modules/esbuild/lib/main.d.ts 1:7
[bundle:bin   ] Module parse failed: Unexpected token (1:7)
[bundle:bin   ] You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
[bundle:bin   ] > export type Platform = 'browser' | 'node' | 'neutral'
[bundle:bin   ] | export type Format = 'iife' | 'cjs' | 'esm'
[bundle:bin   ] | export type Loader = 'base64' | 'binary' | 'copy' | 'css' | 'dataurl' | 'default' | 'empty' | 'file' | 'js' | 'json' | 'jsx' | 'text' | 'ts' | 'tsx'
[bundle:bin   ]  @ ./node_modules/esbuild/lib/ sync ^.*\/.*$ ./main.d.ts ./main.d
[bundle:bin   ]  @ ./node_modules/esbuild/lib/main.js 1826:14-50
[bundle:bin   ]  @ ./node_modules/esbuild-register/dist/node.js 4622:15-33
[bundle:bin   ]  @ ./bin-src/lib/interpret-require.ts 11:29-66
[bundle:bin   ]  @ ./bin-src/lib/getStorybookMetadata.ts 26:28-58
[bundle:bin   ]  @ ./bin-src/lib/getStorybookInfo.ts 18:31-64
[bundle:bin   ]  @ ./bin-src/tasks/storybookInfo.ts 16:43-77
[bundle:bin   ]  @ ./bin-src/tasks/index.ts 15:40-66
[bundle:bin   ]  @ ./bin-src/main.ts 31:32-50
[bundle:bin   ]  @ ./bin-src/register.js 11:14-36
[bundle:bin   ]
[bundle:bin   ] webpack 5.76.0 compiled with 1 error and 16 warnings in 14437 ms
```

This is on the bundle:bin hook after installing esbuild and esbuild-register

[the remark this answers is not in the document]
Comment (ndelangen) [reply bin-src/lib/interpret-require.ts:12]: @JonathanKolnik Let's pair when you have time?

**Comment by JonathanKolnik** [comment]
Apologies for the delay @tmeasday, please test `6.17.5-canary.0`

**Comment by JonathanKolnik** [comment]
@thafryer @tmeasday bumping this!

**Pull request merged**